icicle/tests/test_field_api.cpp

```diff
@@ -120,41 +135,353 @@ TYPED_TEST(FieldApiTest, vectorOps)
   // run(s_reference_target, out_ref.get(), false /*=measure*/, 16 /*=iters*/);
   // run(s_main_target, out_main.get(), false /*=measure*/, 1 /*=iters*/);
 
-  // accumulate
-  auto temp_result = std::make_unique<TypeParam[]>(N);
-  auto initial_in_a = std::make_unique<TypeParam[]>(N);
+  // warmup
+  // run(s_reference_target, out_ref.get(), false /*=measure*/, 16 /*=iters*/);
+  // run(s_main_target, out_main.get(), false /*=measure*/, 1 /*=iters*/);
+
+  // Element-wise vector operations
+  // If config.batch_size>1, (columns_batch=true or false) the operation is done element-wise anyway, so it doesn't
+  // affect the test
 
-  std::memcpy(initial_in_a.get(), in_a.get(), N * sizeof(TypeParam));
-  run(s_reference_target, nullptr, VERBOSE /*=measure*/, vector_accumulate_wrapper, "vector accumulate", ITERS);
-  std::memcpy(temp_result.get(), in_a.get(), N * sizeof(TypeParam));
-  std::memcpy(in_a.get(), initial_in_a.get(), N * sizeof(TypeParam));
+  // // add
+  FieldApiTest<TypeParam>::random_samples(in_a.get(), total_size);
+  FieldApiTest<TypeParam>::random_samples(in_b.get(), total_size);
+  if (!s_is_cuda_registered) {
```

Comment:
I think we can simplify and assume we always have 2 devices to compare against.
Also mentioning cuda specifically in every test will be quite annoying when we have another backend (like metal).